execution/engine/eks_engine.go

```diff
@@ -258,19 +259,31 @@ func (ee *EKSExecutionEngine) FetchUpdateStatus(run state.Run) (state.Run, error
 		return run, err
 	}
 
-	var podList *v1.PodList
-	var pod *v1.Pod
-	// Only fetch associated Pod if the the run doesn't have a podName or if the job exited.
-	if job.Status.Failed == 1 || run.PodName == nil {
-		podList, err = ee.getPodList(run)
-		if err == nil && podList != nil && podList.Items != nil && len(podList.Items) > 0 {
-			pod = &podList.Items[len(podList.Items)-1]
+	var mostRecentPod *v1.Pod
+	var mostRecentPodCreationTimestamp metav1.Time
 
-			if run.PodName == nil && pod != nil {
-				run.PodName = &pod.Name
+	podList, err := ee.getPodList(run)
+
+	if err == nil && podList != nil && podList.Items != nil && len(podList.Items) > 0 {
+		ee.log.Log("message", "iterating over pods")
```

Comment:
might be useful to put in `len(podlist.Items)` in the log line, we can then track how many time are these happening. 
And something to associate it in your statsd PR